cartographer/mapping/map_builder_test.cc

```diff
@@ -31,6 +31,37 @@ namespace cartographer {
 namespace mapping {
 namespace {
 
+const std::string kRangeSensorId = "range";
+const std::string kIMUSensorId = "imu";
+const double kDuration = 2.;
+const double kTimeStep = 0.1;
+
+struct FakeRangeMeasurement {
+  common::Time time;
+  sensor::TimedPointCloud ranges;
+};
+
+std::vector<FakeRangeMeasurement> GenerateFakeRangeMeasurements() {
+  std::vector<FakeRangeMeasurement> measurements;
+  sensor::TimedPointCloud point_cloud;
+  for (double angle = 0.; angle < 1. * M_PI; angle += 0.01) {
+    const double kRadius = 5;
+    point_cloud.emplace_back(kRadius * std::cos(angle),
+                             kRadius * std::sin(angle), 0., 0.);
+  }
+  for (double elapsed_time = 0.; elapsed_time < kDuration;
+       elapsed_time += kTimeStep) {
+    common::Time time =
+        common::FromUniversal(123) + common::FromSeconds(elapsed_time);
```

Comment:
Is it important for the experiment to start at 123? Why not omit this entirely?

Reply:
It is necessary to give an absolute time (FromSeconds only outputs a duration), and I prefer to choose something different from zero.